awx/main/models/events.py

```diff
@@ -352,9 +352,19 @@ def save(self, *args, **kwargs):
         if hasattr(self, 'job') and not from_parent_update:
             if getattr(settings, 'CAPTURE_JOB_EVENT_HOSTS', False):
                 self._update_hosts()
-            if self.event == 'playbook_on_stats':
-                self._update_parents_failed_and_changed()
+            if any([
```

Comment:
I think we can shortcut this entirely by just checking `self.changed` and `self.failed` for `True` and then falling into the JobEvent filter.

We'll save potentially `dozens` of microseconds!

Reply:
Even if this means affecting a parent above the task level I don't think that's such a big deal.